great_expectations/experimental/metric_repository/column_descriptive_metrics_metric_retriever.py

```diff
@@ -25,145 +28,269 @@ class ColumnDescriptiveMetricsMetricRetriever(MetricRetriever):
 
-        assert isinstance(validator.active_batch, Batch)
-        if not isinstance(validator.active_batch, Batch):
-            raise TypeError(
-                f"validator.active_batch is type {type(validator.active_batch).__name__} instead of type {Batch.__name__}"
-            )
+        return metrics
 
-        metric_lookup_key: _MetricKey = (
-            metric_name,
-            tuple(),
-            tuple(),
+    def _get_table_row_count(
+        self, batch_id: str, computed_metrics: dict[_MetricKey, Any]
+    ) -> Metric:
+        metric_name = "table.row_count"
+        value, exception = self._get_metric_from_computed_metrics(
+            metric_name=metric_name,
+            computed_metrics=computed_metrics,
         )
-
-        metrics.append(
-            TableMetric[int](
-                batch_id=validator.active_batch.id,
-                metric_name=metric_name,
-                value=computed_metrics[metric_lookup_key],
-                exception=None,  # TODO: Pass through a MetricException() if an exception is thrown
-            )
+        return TableMetric[int](
+            batch_id=batch_id,
+            metric_name=metric_name,
+            value=value,
+            exception=exception,
         )
 
+    def _get_table_columns(
+        self, batch_id: str, computed_metrics: dict[_MetricKey, Any]
+    ) -> Metric:
         metric_name = "table.columns"
-        metric_lookup_key = (metric_name, tuple(), tuple())
-        metrics.append(
-            TableMetric[List[str]](
-                batch_id=validator.active_batch.id,
-                metric_name=metric_name,
-                value=computed_metrics[metric_lookup_key],
-                exception=None,  # TODO: Pass through a MetricException() if an exception is thrown
-            )
+        value, exception = self._get_metric_from_computed_metrics(
+            metric_name=metric_name,
+            computed_metrics=computed_metrics,
+        )
+        return TableMetric[List[str]](
+            batch_id=batch_id,
+            metric_name=metric_name,
+            value=value,
+            exception=exception,
         )
 
+    def _get_table_column_types(
+        self, batch_id: str, computed_metrics: dict[_MetricKey, Any]
+    ) -> Metric:
         metric_name = "table.column_types"
-        metric_lookup_key = (metric_name, tuple(), "include_nested=True")
-
-        raw_column_types: list[dict[str, Any]] = computed_metrics[metric_lookup_key]  # type: ignore[assignment] # Metric results from computed_metrics are not typed
-
+        metric_lookup_key: _MetricKey = (metric_name, tuple(), "include_nested=True")
+        value, exception = self._get_metric_from_computed_metrics(
+            metric_name=metric_name,
+            metric_lookup_key=metric_lookup_key,
+            computed_metrics=computed_metrics,
+        )
+        raw_column_types: list[dict[str, Any]] = value
         column_types_converted_to_str: list[dict[str, str]] = [
```

Comment:
Extra handling is needed, since it can come back as objects (ie. Pandas.int64 or something similar)